fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/api/registry/LandPathNodeTypesRegistry.java

```diff
@@ -0,0 +1,277 @@
+		Objects.requireNonNull(state, "BlockState cannot be null!");
+		Objects.requireNonNull(world, "BlockView cannot be null!");
+		Objects.requireNonNull(pos, "BlockPos cannot be null!");
+
+		// Gets the node type provider for the block.
+		PathNodeTypeProvider provider = NODE_TYPES.get(state.getBlock());
+
+		//Gets the node type from the registered provider.
+		if (provider instanceof DynamicPathNodeTypeProvider dynamicProvider) {
+			//This can return null too.
+			return dynamicProvider.getPathNodeType(state, world, pos, neighbor);
+		} else if (provider instanceof StaticPathNodeTypeProvider staticProvider) {
+			//This can return null too.
+			return staticProvider.getPathNodeType(state, neighbor);
+		}
+
+		//If no valid provider is found returns null.
+		return null;
+	}
+
+	/**
+	 * Gets the {@link PathNodeType} from the provider registered for the specified block state.
+	 *
+	 * <p>This searches only for static providers.
+	 *
+	 * <p>If no valid {@link PathNodeType} provider is registered for the block, it returns {@code null}.
+	 * You cannot use this method to retrieve vanilla block node types.
+	 *
+	 * @param state    Current block state.
+	 * @param neighbor Specifies if the block is not a directly targeted block, but a neighbor block in the path.
+	 * @return the custom {@link PathNodeType} from the static provider registered for the specified block,
+	 * passing the block state to the provider, or {@code null} if no valid static provider is registered for the block.
+	 */
+	@Nullable
+	public static PathNodeType getStaticPathNodeType(BlockState state, boolean neighbor) {
+		Objects.requireNonNull(state, "BlockState cannot be null!");
+
+		// Gets the node type provider for the block.
+		PathNodeTypeProvider provider = NODE_TYPES.get(state.getBlock());
+
+		//Gets the node type from the registered provider.
+		if (provider instanceof StaticPathNodeTypeProvider staticProvider) {
+			//This can return null too.
+			return staticProvider.getPathNodeType(state, neighbor);
+		}
+
+		//If no valid provider is found returns null.
+		return null;
+	}
+
+	/**
+	 * Gets the provider type registered for the specified block.
+	 *
+	 * @param block Current block.
+	 * @return an enum constant indicating the provider type registered for the specified block.
+	 */
+	public static ProviderType getProviderType(Block block) {
+		Objects.requireNonNull(block, "Block cannot be null!");
+
+		// Gets the node type provider for the block.
+		PathNodeTypeProvider provider = NODE_TYPES.get(block);
+
+		//Gets the node type from the registered provider.
+		if (provider instanceof StaticPathNodeTypeProvider) {
+			return ProviderType.STATIC;
+		} else if (provider instanceof DynamicPathNodeTypeProvider) {
+			return ProviderType.DYNAMIC;
+		}
+
+		//If no valid provider is found returns NONE.
+		return ProviderType.NONE;
+	}
+
+	/**
+	 * General provider, this is a marker interface.
+	 */
+	private interface PathNodeTypeProvider {
```

Comment:
Would it be sensible to seal this interface?

Reply:
Done.
And i learned a new thing. 👍